Add support for lightmaps in ogre2

iche033 opened the pull request:
depends on ignitionrobotics/ign-common#132

main changes:
* updated mesh factory to create mesh with multiple texture coordinate sets
* added lightmap APIs to Material - users can choose which texture coordinate set to use for the lightmap
* To keep things simple, lightmap is currently blended with the base albedo map using the Overlay bend mode (here's a [description](https://helpx.adobe.com/photoshop/using/blending-modes.html) of the different blend modes).
* fixed a transparency issue that happens when a grayscale albedo map is used.

chapulina commented:
@osrf-jenkins run tests please :pray:

chapulina commented:
Ah no, triggered too soon :woman_facepalming:

chapulina commented:
I fixed code check and a failing test on 1d69e73.

@iche033 , I removed a big chunk of code from the test which didn't seem to be used. Please check if that was indeed unnecessary and if it was ok to remove it.

iche033 commented:
> I removed a big chunk of code from the test which didn't seem to be used. Please check if that was indeed unnecessary and if it was ok to remove it.

actually the test was supposed to test and verify copying from this `common::Material` but it was doing a `CopyFrom` from the wrong material variable. I've updated the test in 0d62019.
